File: 0125-valid-palindrome/0125-valid-palindrome.swift
class Solution {
    func isPalindrome(_ s: String) -> Bool {
        let cleaningString = s.lowercased().filter {$0.isNumber || $0.isLetter}
        let characters = Array(cleaningString)

        var left = 0
        var right = cleaningString.count - 1
        while left < right{
            if characters[left] != characters[right]{
                return false
            }
            left += 1
            right -= 1
        }
        return true;
    }
}

File: 0125-valid-palindrome/0125-valid-palindrome.py
import re

class Solution:
    def isPalindrome(self, s: str) -> bool:
        s = re.sub(pattern= '[^a-zA-Z0-9]',repl='', string = s).lower()
        left = 0
        right = len(s) - 1
        while(left <= right):
            if s[left] != s[right]:
                return False
            else:
                left += 1
                right -= 1
        return True